scipy/stats/tests/test_stats.py

```diff
@@ -2253,6 +2253,96 @@ def test_ttest_ind():
         np.seterr(**olderr)
 
 
+def test_ttest_ind_permutations():
+    # Test on horizontal dimension
+    N = 20
+    np.random.seed(0)
+    a = np.vstack((np.arange(3*N//4), np.random.random(3*N//4)))
+    b = np.vstack((np.arange(N//4) + 100, np.random.random(N//4)))
+    p_t_stats, pvalues = stats.ttest_ind(a, b, axis=1, equal_var=False)
+    np_t_stats, pvalues = stats.ttest_ind(a, b, axis=1, equal_var=False,
+                                          permutations=1000,
+                                          random_state=0)
+
+    assert_array_almost_equal(p_t_stats, np_t_stats, 5)
+    assert_array_almost_equal(pvalues, array([0.000999, 0.69031]))
+
+    # Test on vertical dimension
+    N = 20
+    np.random.seed(0)
+    a = np.vstack((np.arange((3*N)//4), np.random.random(3*N//4))).transpose()
+    b = np.vstack((np.arange(N//4) + 100, np.random.random(N//4))).transpose()
+    p_t_stats, pvalues = stats.ttest_ind(a, b, axis=0, equal_var=False)
+    np_t_stats, pvalues = stats.ttest_ind(a, b, axis=0, equal_var=False,
+                                          permutations=1000,
+                                          random_state=0)
+
+    assert_array_almost_equal(p_t_stats, np_t_stats, 5)
+    assert_array_almost_equal(pvalues, array([0.000999, 0.69031]))
+
+    # Test on 1 dimensional case
+    N = 20
+    np.random.seed(0)
+    a = np.arange(3*N//4)
+    b = np.arange(N//4) + 100
+    p_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=False)
+    np_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=False,
+                                          permutations=1000,
+                                          random_state=0)
+
+    assert_array_almost_equal(p_t_stats, np_t_stats, 5)
+    assert_array_almost_equal(pvalues, 0.000999)
+
+    # Test just arrays
+    N = 20
+    np.random.seed(0)
+    a = range(3*N//4)
+    b = range(100, N//4 + 100)
+    p_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=False)
+    np_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=False,
+                                          permutations=1000,
+                                          random_state=0)
+
+    assert_array_almost_equal(p_t_stats, np_t_stats, 5)
+    assert_array_almost_equal(pvalues, 0.000999)
+
+    # Test equal variance
+    np.random.seed(0)
+    a = np.arange(10)
+    b = np.arange(10) + 100
+    p_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=True)
+    np_t_stats, pvalues = stats.ttest_ind(a, b, equal_var=True,
+                                          permutations=1000,
+                                          random_state=0)
+    assert_array_almost_equal(p_t_stats, np_t_stats, 5)
+    assert_array_almost_equal(pvalues, 0.000999)
+
+    # Test out random seed
+    N = 20
+    seed = np.random.RandomState(seed=0)
```

Comment:
newer versions of numpy prefer `Generator`

suggest to swap to `Generator` and add a `skipif` pytest annotation if less than `1.18.0` via `NumpyVersion`

Reply:
@rlucas7 If it's OK, I'm going to leave this as-is for now so that the test runs even with old versions of NumPy. When we stop supporting old versions, maybe that's a good time to start using `Generator`. Does that make sense?